bazel/lib/dependabot/bazel/file_parser/starlark_parser.rb

```diff
@@ -0,0 +1,360 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "dependabot/bazel/file_parser"
+
+module Dependabot
+  module Bazel
+    class FileParser
+      class StarlarkParser
+        extend T::Sig
+
+        class FunctionCall < T::Struct
+          const :name, String
+          const :arguments, T::Hash[String, T.untyped]
+          const :positional_arguments, T::Array[T.untyped]
+          const :line, Integer
+        end
+
+        sig { params(content: String).void }
+        def initialize(content)
+          @content = content
+          @position = T.let(0, Integer)
+          @line = T.let(1, Integer)
+          @length = T.let(content.length, Integer)
+        end
+
+        sig { returns(T::Array[FunctionCall]) }
+        def parse_function_calls
+          function_calls = T.let([], T::Array[FunctionCall])
+
+          while @position < @length
+            skip_whitespace_and_comments
+
+            next unless @position < @length
+
+            start_position = @position
+            function_call = try_parse_function_call
+            function_calls << function_call if function_call
+
+            advance if @position == start_position
+          end
+
+          function_calls
+        end
+
+        private
+
+        sig { returns(T.nilable(String)) }
+        def current_char
+          return nil if @position >= @length
+
+          @content[@position]
+        end
+
+        sig { params(offset: Integer).returns(T.nilable(String)) }
+        def peek_char(offset = 1)
+          peek_pos = @position + offset
+          return nil if peek_pos >= @length
+
+          @content[peek_pos]
+        end
+
+        sig { void }
+        def advance
```

Comment:
A more descriptive name would help the reader understand what this method does each time it is called. However, I'm not sure exactly what name. Some options could be:

- `advance_char`
- `next_char`
- `move_to_next_char`
- `move_to_next_char_or_line`

Or something along these lines.